core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -677,12 +672,26 @@ public TypeProvider getTypes()
 
         public LocalDynamicFiltersCollector getDynamicFiltersCollector()
         {
-            return dynamicFiltersCollector;
+            return taskContext.getLocalDynamicFiltersCollector();
         }
 
         private void addLocalDynamicFilters(Map<DynamicFilterId, Domain> dynamicTupleDomain)
         {
-            dynamicFiltersCollector.collectDynamicFilterDomains(dynamicTupleDomain);
+            taskContext.addDynamicFilter(dynamicTupleDomain);
+        }
+
+        private void registerCoordinatorDynamicFilters(List<DynamicFilters.Descriptor> dynamicFilters)
+        {
+            if (!isEnableDistributeDynamicFilters(taskContext.getSession())) {
+                return;
+            }
+            Set<DynamicFilterId> consumedFilterIds = dynamicFilters.stream()
+                    .map(DynamicFilters.Descriptor::getId)
+                    .collect(toImmutableSet());
+            // Don't wait for coordinator to send a dynamic filter which is only collected locally
+            LocalDynamicFiltersCollector dynamicFiltersCollector = taskContext.getLocalDynamicFiltersCollector();
+            dynamicFiltersCollector.register(
```

Comment:
this will fail if there are 2 table scans (e.g colocated joins) that consume same DF (`io.trino.sql.planner.LocalDynamicFiltersCollector#register` prevents duplicate registration)

Reply:
would we have duplicate registration though ?
i'm doing `difference(consumedFilterIds, dynamicFiltersCollector.getRegisteredDynamicFilterIds())`